framework/decode/vulkan_offscreen_swapchain.cpp

```diff
@@ -222,7 +276,36 @@ VkResult VulkanOffscreenSwapchain::QueuePresentKHR(VkResult
                                                    const QueueInfo*                      queue_info,
                                                    const VkPresentInfoKHR*               present_info)
 {
-    if (present_info->waitSemaphoreCount > 0)
+    if (insert_frame_boundary_ && command_buffer_ != VK_NULL_HANDLE)
```

Comment:
Yes, but the command buffer creation can fail. If it is the case (and that we're in release mode, because there are asserts) then it's better to just not randomly crash in a segfault here and have the insertion ignored...